shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -283,11 +284,15 @@ private CharSequence getClipboardData(PlatformChannel.ClipboardContentFormat for
 
     if (!clipboard.hasPrimaryClip()) return null;
 
-    ClipData clip = clipboard.getPrimaryClip();
-    if (clip == null) return null;
-
-    if (format == null || format == PlatformChannel.ClipboardContentFormat.PLAIN_TEXT) {
-      return clip.getItemAt(0).coerceToText(activity);
+    try {
+      ClipData clip = clipboard.getPrimaryClip();
+      if (clip == null) return null;
+      if (format == null || format == PlatformChannel.ClipboardContentFormat.PLAIN_TEXT) {
+        return clip.getItemAt(0).coerceToText(activity);
+      }
+    } catch (SecurityException e) {
+      Log.w("flutter", "ClipData", e);
```

Comment:
Normally, we define a `private static final String TAG = "PlatformPlugin";` in the class.

Can we guide users a little better here? 

For example, instead of just saying "ClipData", maybe something like:

```
Log.w(TAG, "Attempted to get clipboard data that requires additional permission(s). See the exception details for which permission(s) are required, and consider adding them to your Android Manifest as described in https://developer.android.com/guide/topics/permissions/overview", e);
```

With some approrpriate breaking of the string for line length :)